controller/bluegreen.go

```diff
@@ -20,73 +21,73 @@ func (c *Controller) rolloutBlueGreen(r *v1alpha1.Rollout, rsList []*appsv1.Repl
 	logCtx := logutil.WithRollout(r)
 	newRS, oldRSs, err := c.getAllReplicaSetsAndSyncRevision(r, rsList, true)
 	if err != nil {
-		return err
+		return c.metricsServer.IncError(r, err)
```

Comment:
Yeah, I agree that makes sense.  One side effort of this approach is that I will need to add an extra counter for reconciliation errors because I don't have the rollout variable to get the required strategy label at https://github.com/argoproj/argo-rollouts/blob/master/controller/controller.go#L223.  I could use the informer to get the rollout, but that feels like the controller is doing too much work during the error handling. However, I don't think that side effort is bad though. The new counter will only be missing the strategy field, and in this case, the operator will need to check the controller's logs for more information. What do you think?

Reply:
I think both approaches are acceptable.  Please choose one which you like the best.